streams/src/main/java/org/apache/kafka/streams/state/internals/StoreQueryUtils.java

```diff
@@ -142,7 +142,9 @@ public static void updatePosition(
 
         if (stateStoreContext != null && stateStoreContext.recordMetadata().isPresent()) {
             final RecordMetadata meta = stateStoreContext.recordMetadata().get();
-            position.withComponent(meta.topic(), meta.partition(), meta.offset());
+            if (meta.topic() != null) {
```

Comment:
Fixes some NPEs from the unit tests. I figure it's fine to have this here, since there's technically no contract that says topic can't be null and we're already guarding against a null context and missing recordMetadata.